commit 5fd23f7e1d74c0fd100ffb0b04dc85727760d9ea
Author: Chuck Lever <chuck.lever@oracle.com>
Date:   Thu Jun 11 13:47:10 2015 -0400

    SUNRPC: Address kbuild warning in net/sunrpc/debugfs.c
    
    Cross-compile test on ARCH=mn10300:
    
    In file included from include/linux/list.h:8:0,
                     from include/linux/wait.h:6,
                     from include/linux/fs.h:6,
                     from include/linux/debugfs.h:18,
                     from net/sunrpc/debugfs.c:7:
    net/sunrpc/debugfs.c: In function 'fault_disconnect_write':
    include/linux/kernel.h:723:17: warning: comparison of distinct pointer
    types lacks a cast
        (void) (&_min1 == &_min2);  \
                       ^
    >> net/sunrpc/debugfs.c:307:8: note: in expansion of macro 'min'
        len = min(len, sizeof(buffer) - 1);
    
    Fixes: ('SUNRPC: Transport fault injection')
    Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
    Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>

diff --git a/net/sunrpc/debugfs.c b/net/sunrpc/debugfs.c
index 7cc1b8a6ef6d..e7b4d93566df 100644
--- a/net/sunrpc/debugfs.c
+++ b/net/sunrpc/debugfs.c
@@ -304,7 +304,8 @@ fault_disconnect_write(struct file *filp, const char __user *user_buf,
 {
 	char buffer[16];
 
-	len = min(len, sizeof(buffer) - 1);
+	if (len >= sizeof(buffer))
+		len = sizeof(buffer) - 1;
 	if (copy_from_user(buffer, user_buf, len))
 		return -EFAULT;
 	buffer[len] = '\0';